File: 1584-min-cost-to-connect-all-points/1584-min-cost-to-connect-all-points.cpp
// Prim's algorithm // timeO(v^2)
class Solution {
public:
    int weight(vector<int>& a, vector<int>& b){
        return abs(a[0]-b[0]) + abs(a[1]-b[1]);
    }
    int minCostConnectPoints(vector<vector<int>>& points) {
        int v = points.size();
        vector<int> key(v, INT_MAX);
        vector<bool> inMST(v,false);
        // 
        priority_queue<pair<int,int>, vector<pair<int,int>>, greater<pair<int,int>> > pq; // key - vertex
        pq.push({0,0});
        key[0] = 0;
        while(!pq.empty()){
            int curr_vertex = pq.top().second; pq.pop();
            if(inMST[curr_vertex]) continue;
            inMST[curr_vertex] = true;
            for(int i =0; i<v; i++){
                int adj_vertex = i;
                int w = weight(points[curr_vertex],points[adj_vertex]);
                if(!inMST[i] && w < key[adj_vertex]){
                    key[adj_vertex] = w;
                    pq.push({w,adj_vertex});
                }
            }
        }
        int ans = 0;
        for(int& i : key) ans += i;
        return ans;
    }
};

File: 1584-min-cost-to-connect-all-points/1584-min-cost-to-connect-all-points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var minCostConnectPoints = function(points) {
    const v = points.length;
    const find = (x) =>{
        if(parent[x] === x) return x;
        return find(parent[x]);
    }
    const unite = function (a,b) {
        a = find(a);
        b = find(b);
        if(a==b) return false;
        if(size[a]>size[b]){
            parent[b] = a;
            size[a] += size[b];
        }else{
            parent[a] = b;
            size[b] += size[a];
        }
        return true;
    }
    const weight = function (v1,v2) {
        return Math.abs(v1[0]-v2[0]) + Math.abs(v1[1]-v2[1]);
    }
//     Make edge list 
    const edge_lists = [];
    // [ [weight,intial vertex, terminal vertex] ]
    for(let i =0; i<v; i++){
        for(let j = i+1; j<v; j++){
            edge_lists.push([weight(points[i],points[j]),i,j]);
        }
    }
    edge_lists.sort((a,b)=>{return a[0]-b[0]});
    //
    // make set
    const parent = new Array(v);
    for(let i =0; i<v; i++) parent[i] = i;
    const size = new Array(v).fill(1);
    let ans = 0;
    edge_lists.forEach(function(tp){
        if(unite(tp[1],tp[2])) ans += tp[0];      
    });
    return ans;
};